src/main/scala/org/broadinstitute/dsde/workbench/sam/api/StandardUserInfoDirectives.scala

```diff
@@ -114,18 +131,8 @@ object StandardUserInfoDirectives {
   }
 
   private def lookUpByGoogleSubjectId(googleSubjectId: GoogleSubjectId, directoryDAO: DirectoryDAO, samRequestContext: SamRequestContext): IO[SamUser] =
-    for {
-      maybeSamUser <- directoryDAO.loadUserByGoogleSubjectId(googleSubjectId, samRequestContext)
-      samUser <- maybeSamUser match {
-        case Some(samUser) => IO.pure(samUser)
-        case None => directoryDAO.loadSubjectFromGoogleSubjectId(googleSubjectId, samRequestContext).flatMap {
-          case Some(_) =>
-            IO.raiseError(new WorkbenchExceptionWithErrorReport(ErrorReport(StatusCodes.Conflict, s"subjectId $googleSubjectId is not a WorkbenchUser")))
-          case None =>
-            IO.raiseError(new WorkbenchExceptionWithErrorReport(ErrorReport(StatusCodes.Forbidden, s"Google Id $googleSubjectId not found in sam")))
-        }
-      }
-    } yield samUser
+    directoryDAO.loadUserByGoogleSubjectId(googleSubjectId, samRequestContext).map(
+      _.getOrElse(throw new WorkbenchExceptionWithErrorReport(ErrorReport(StatusCodes.Forbidden, s"Google Id $googleSubjectId not found in sam"))))
```

Comment:
I think this works fine but I like to avoid `throw` inside IO map/flatMap because it's a side effect. Could do this instead:
```
    directoryDAO.loadUserByGoogleSubjectId(googleSubjectId, samRequestContext).flatMap { user =>
      IO.fromOption(user)(new WorkbenchExceptionWithErrorReport(ErrorReport(StatusCodes.Forbidden, s"Google Id $googleSubjectId not found in sam")))
```